kratos/modeler/operation/compute_surrogate_boundary_data.cpp

```
@@ -0,0 +1,165 @@
+                    if (!search_result.IsObjectFound()) 
+                    {
+                        search_result = conditions_bin.SearchNearest(point);
+                    }
+                    if(search_result.IsObjectFound()) // It should always be found
+                    {
+                        auto p_result = search_result.Get().get();
+                        GeometryPtrType geometry = p_result->pGetGeometry();
+                        const PointsArrayType& points = geometry->Points();
+                        Point closest_point = NearestPointUtilities::TriangleNearestPoint(point, points[0],points[1],points[2]);
+                        
+                        for (std::size_t ii = 0; ii < 3; ii++) 
+                        {
+                            surrogate_boundary_data[node_index].GetVectorDistance()[ii] = point[ii] - closest_point[ii];
+                        }
+                    } else {
+                        KRATOS_WARNING("SurrogateBoundaryModeler") << "Input geometry has no elements or conditions. Unable to compute distance to skin." << std::endl;
+                    } 
+                    
+                    double d = norm_2(surrogate_boundary_data[node_index].GetVectorDistance());
+                    if (colors.GetNodalColor(i,j,k) == inside_color) 
+                    {
+                        surrogate_boundary_data[node_index].GetSignedDistance() = -d;
+                    } else {
+                        surrogate_boundary_data[node_index].GetSignedDistance() = d;
+                    }
+                }
+            } 
+        }  
+    }
+
+    std::ofstream output_file(output_path);
+    
+    // Check if the file was opened successfully
+    if (output_file.is_open()) {
+        output_file <<  PrintSBData(surrogate_boundary_data);
+        output_file.close();
+    } else  
+        KRATOS_WARNING("ComputeSurrogateBoundaryData") << "Data could not be saved in file" << std::endl;
```

As we did it, the operation only makes sense if the data is written to a file. In other words, if not successfully saved, al the calculations we did are lost when we leave the `Execute` scope.

Considering this, I'd throw an error in here rather than a warning.

---

Agree